File: workshop.c
#include <stdio.h>
#include <fcntl.h>
#include <unistd.h>

int main() {
    int fd;
    char buffer[128];
    int bytes;
    
    // System Call: open
    fd = open("sample.txt", O_RDONLY);
    if (fd == -1) return 1;

    // System Call: read
    bytes = read(fd, buffer, sizeof(buffer));

    // System Call: write (1 = stdout)
    write(1, buffer, bytes);

    // System Call: close
    close(fd);
    return 0;
}

File: workshop.py
import os
# System Call: open
fd = os.open("sample.txt", os.O_RDONLY)
# System Call: read
data = os.read(fd, 128)
# System Call: write (1 = stdout)
os.write(1, data)
# System Call: close
os.close(fd)
